CMakeLists.txt

```diff
@@ -666,6 +666,7 @@ set (CMAKE_MODULE_PATH ${HDF_RESOURCES_DIR} ${CMAKE_MODULE_PATH})
 # Option to Enable HDFS
 #-----------------------------------------------------------------------------
 option (HDF5_ENABLE_HDFS "Enable HDFS" OFF)
+set(H5_HAVE_LIBHDFS_SETTING OFF)
```

Comment:
Rather than adding a bunch of new variables for all the settings that could be ON or OFF, we should probably just force the option variable values to OFF if the feature can't be enabled and then just use the value of the option variables.

Reply:
Agree, options are ON/OFF and should be used in the settings file (if the option is requested and fails to be allowed the option should be set to OFF and a message displayed). H5_* variable should be only for code.

Reply:
If there is not an option for a setting variable then we can create a temp at the time of creating the settings file.

Reply:
I'll update it.